dvc/repo/add.py

```diff
@@ -2,6 +2,22 @@
 from dvc.stage import Stage
 
 
+def cleanup_gitignore(add_method):
+    def run(*args, **kw):
+        repo = args[0]
```

Comment:
Also, probably worth adding an assert that would make sure that repo is actually a Repo() instance.